### PR TITLE
exit job script when GEOSldas.x run failed

### PR DESCRIPTION
**Pull request by weiyuan-jiang**
exit when run failed.

### CODE REVIEW

File: src/Applications/LDAS_App/lenkf.j.template [hunk 1]
```diff
@@ -357,10 +357,13 @@ while ( $counter <= ${NUM_SGMT} )
    
    if( -e EGRESS.ldas ) then
       set rc = 0
+      echo GEOSldas Run Status: $rc
    else
       set rc = -1
+      echo GEOSldas Run Status: $rc
+      echo "GEOSldas run failed, exit without post-processing"
+      exit
```
Comment (gmao-rreichle) [inline src/Applications/LDAS_App/lenkf.j.template:365]: Do we really want to exit here?  If we do, there will not  be an attempt at post-processing.  I would remove the "exit" here and in the preceding line say something like: 
`echo "GEOSldas run FAILED, attempting post-processing..."`
Then also move the "if ... exit" statement to occur ~line 714 (see comment below).

File: src/Applications/LDAS_App/lenkf.j.template [hunk 2]
```diff
@@ -732,6 +735,11 @@ EOF
 ## End of the while ( $counter <= ${NUM_SGMT} ) loop ##
 end
 
+if ($rc != 0) then
```
Comment (gmao-rreichle) [inline src/Applications/LDAS_App/lenkf.j.template:738]: If we drop the exit statement above, this if block should be moved to ~line 714, right?

File: src/Applications/LDAS_App/lenkf.j.template (hunk 1 above)
Comment (weiyuan-jiang) [reply src/Applications/LDAS_App/lenkf.j.template:365]: This one should be "break" which comes out of the loop

File: src/Applications/LDAS_App/lenkf.j.template [hunk 3]
```diff
@@ -357,10 +357,13 @@ while ( $counter <= ${NUM_SGMT} )
    
    if( -e EGRESS.ldas ) then
       set rc = 0
+      echo GEOSldas Run Status: $rc
    else
       set rc = -1
+      echo GEOSldas Run Status: $rc
+      echo "GEOSldas run failed, exit without post-processing"
+      break
```
Comment (gmao-rreichle) [inline src/Applications/LDAS_App/lenkf.j.template:365]: If we "break" here, don't we then immediately hit the "exit" statement in line 740? If this is the desired behavior, then we might as well "exit" right here, or not? 
My earlier thought was that we should try to finish out the current SGMT, and then have the if block with the rc-triggered exit in ~line 714, but on 2nd thought I'm not sure this makes much sense.
In summary, change "break" back to "exit" here and remove the new if block that was added ~Line 738?

**Review by gmao-rreichle — APPROVED** [review]

**Pull request merged**